File: 2/2000s/28s/2873_MaximumValueOfAnOrderedTripletI.cpp
/*
https://leetcode.com/problems/maximum-value-of-an-ordered-triplet-i/description/?envType=daily-question&envId=2025-04-02
*/

class Solution {
public:
    long long maximumTripletValue(vector<int>& nums) {
        int sz=nums.size();
        long long res=LONG_MIN;
        for(int i=0; i<sz; ++i){
            if(i+1<=sz-1){
                for(int j=i+1; j<sz; ++j){
                    if(j+1<=sz-1){
                        for(int k=j+1; k<sz; ++k){
                            long long tochk=(nums[i]-nums[j])*(long long)nums[k];
                            if(tochk>=0) res=max(res, tochk);
                        }
                    }
                }
            }
        }
        return res!=LONG_MIN?res:0;
    }
};

File: 2/2000s/28s/2873/2873_MaximumValueOfAnOrderedTripletI.py
# https://leetcode.com/problems/maximum-value-of-an-ordered-triplet-i/description/?envType=daily-question&envId=2025-04-02

class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        lengthOfNums=len(nums)
        longMinValue=-9223372036854775808
        resultantMaxValue=longMinValue
        for i in range(0, lengthOfNums, 1):
            if i+1<=lengthOfNums-1:
                for j in range(i+1, lengthOfNums, 1):
                    if j+1<=lengthOfNums-1:
                        for k in range(j+1, lengthOfNums, 1):
                            currentTripletValue=(nums[i]-nums[j])*nums[k]
                            if currentTripletValue>=0:
                                resultantMaxValue=max(resultantMaxValue, currentTripletValue)
        if resultantMaxValue!=longMinValue:
            return resultantMaxValue
        return 0
